### PR TITLE
Add SAI_PORT_ATTR_EYE_VALUES

### PR DESCRIPTION
**Pull request by nikos-li**

### CODE REVIEW

**Comment by kcudnik** [comment]
A new value type SAI_ATTR_VALUE_TYPE_PORT_EYE_VALUES_LIST must be added after this: https://github.com/opencomputeproject/SAI/blob/master/meta/saimetadatatypes.h#L360

File: inc/saitypes.h [hunk 1]
```diff
@@ -691,6 +691,40 @@ typedef struct _sai_segment_list_t
...
+ * must set the count member to the actual number of objects filled in the
+ * list member and return #SAI_STATUS_BUFFER_OVERFLOW. Once the caller
+ * gets such a return code, it may use the returned count member to
+ * re-allocate the list and retry.
+ */
+typedef struct _sai_port_eye_values_list_t
+{
+    uint32_t count;
+    sai_port_lane_eye_values_t *list;
+} sai_port_eye_values_list_t;
```
Comment (kcudnik) [inline inc/saitypes.h]: proposal to change name to sai_port_lane_eye_values_list_t or remove "lane" from name sai_port_lane_eye_values_t

File: inc/saitypes.h [hunk 2]
```diff
@@ -691,6 +691,40 @@ typedef struct _sai_segment_list_t
     sai_ip6_t *list;
 } sai_segment_list_t;
 
+/**
+ * @brief Defines a lane with its eye values
+ */
+typedef struct _sai_port_lane_eye_values_t
+{
+    uint32_t lane_number;
```
Comment (kcudnik) [inline inc/saitypes.h]: would "lane" be enough ? you can add comment to that filed that thie is lane number, but single "lane" in this context is probably self explaining

Comment (nikos-li) [reply inc/saitypes.h]: I'm fine with just lane.

File: inc/saitypes.h (hunk 1 above)
Comment (nikos-li) [reply inc/saitypes.h]: sai_port_lane_eye_values_list_t can imply the values for one lane which is not the case (despite the list in the name). I can do sai_port_lanes_eye_values_list_t if you like - I don't want any confusion on such APIs.

**Comment by nikos-li** [comment]
>A new value type SAI_ATTR_VALUE_TYPE_PORT_EYE_VALUES_LIST must be added after this: >https://github.com/opencomputeproject/SAI/blob/master/meta/saimetadatatypes.h#L360

Correct. I did a git add . and was in the wrong directory and missed that it didn't pick that change. Will correct tomorrow.

File: inc/saitypes.h (hunk 2 above)
Comment (lguohan) [reply inc/saitypes.h]: what is the unit for these for left/right/up/down values?

**Comment by nikos-li** [comment]
@lguohan Please assign someone to fix the SAI dictionary unless I'm misreading the build error.

**Comment by lguohan** [comment]
@nikos-li , can you take a look at this PR and add word to dictionary? https://github.com/opencomputeproject/SAI/pull/776/files

**Comment by kcudnik** [comment]
please add word "millivolts" to meta/aspell.en.pws

**Comment by nikos-li** [comment]
@lguohan @kcudnik meta/acronyms.txt is not allowing mV acronym. If acronyms.txt is only allowing all capital acronyms then that's wrong and needs to be corrected.

File: inc/saitypes.h [hunk 3]
```diff
@@ -691,6 +691,41 @@ typedef struct _sai_segment_list_t
...
 
+/**
+ * @brief Defines a lane with its eye values with the values being in
+ * millivolts.
+ */
+typedef struct _sai_port_lane_eye_values_t
+{
+    uint32_t lane;
+    int32_t left;
+    int32_t right;
```
Comment (SAI-shared-user) [inline inc/saitypes.h]: left, right seems to be related to time instead of voltage.

**Comment by lguohan** [comment]
one thing we can do is that use millivolts instead of mV here.

**Comment by nikos-li** [comment]
@kcudnik @lguohan How do I add mV and mUI in acronyms.txt? It seems to accept only all upper case and that is not right for those units. What else needs to change for acronyms to accept mixed lower/upper case?

**Comment by kcudnik** [comment]
add mV to meta/aspell.en.pws

File: inc/saitypes.h [hunk 4]
```diff
@@ -691,6 +691,41 @@ typedef struct _sai_segment_list_t
     sai_ip6_t *list;
 } sai_segment_list_t;
 
+/**
+ * @brief Defines a lane with its eye values with the up and down values
+ * being in mV and left and right being in mUI.
```
Comment (lguohan) [inline inc/saitypes.h]: I am not sure about mUI. https://en.wikipedia.org/wiki/Unit_interval_(data_transmission)

I think it should be UI.

Comment (nikos-li) [reply inc/saitypes.h]: UI with int32_t will cause rounding up/down. mUI is correct.

Comment (lguohan) [reply inc/saitypes.h]: what does 'm' stands for here? milli UI, like UI/1000?

Comment (nikos-li) [reply inc/saitypes.h]: Yes. m in measurements stands for milli.

**Pull request merged**